lib/route.js

```diff
@@ -307,12 +307,28 @@ function buildRouting (options) {
         constraints.version = opts.version
       }
 
-      const headHandler = router.find('HEAD', opts.url, constraints)
-      const hasHEADHandler = headHandler != null
+      let headHandler = null
+      let hasHEADHandler = false
+
+      if (shouldExposeHead && opts.method === 'GET') {
```

Comment:
cant we put these temporary variables into the if condition?

```suggestion
      if (shouldExposeHead && opts.method === 'GET') {
        let headHandler = null
        let hasHEADHandler = false
```

Reply:
I can move `headHandler` inside the if statement. But `hasHEADHandler` cannot be inside, since it's used afterwards.